engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -336,22 +331,35 @@ public void setJobsAvailableCallback(final Consumer<String> onJobsAvailableCallb
   }
 
   @Override
-  public long findBackedOffJobs(final long timestamp, final BiPredicate<Long, JobRecord> callback) {
-    nextBackOffDueDate = -1L;
+  public ProcessTriggerResult processExpiredTriggersBefore(
+      final long timestamp, final TriggerVisitor<JobRecord> visitor) {
+    final var nextBackOffDueDate = new MutableLong(-1);
+    final var cache = triggerCache.stage();
```

Comment:
:question: Is this the only public method? Needs to check on PC. 

If not, we should hide the others so it is the only way to cache stuff.

Reply:
There are other methods, but I'm sure we can come up with a better API in a second pass :+1: